### PR TITLE
Adding the `ignore_armour` boolean to `DamagePlayer` and `InflictDama…

### PR DESCRIPTION
**Pull request by Jeroen52**
…ge` to do damage straight to health, ignoring armour.

Please do check if the code is sane, I have my doubts that [this](https://github.com/oscar-broman/samp-weapon-config/compare/master...Jeroen52:master#diff-bb9c2a9539291b896d938b65a38c33e9R1203) is sane.

Made this pull request because of this: http://forum.sa-mp.com/showpost.php?p=3622195&postcount=380

### CODE REVIEW

**Comment by Jeroen52** [comment]
Forgot to commit documentation, which is now done and pushed and added to this pull request.

**Comment by Crayder** [comment]
That would only apply to those three weapon ids...

**Comment by Jeroen52** [comment]
@Crayder: I got a specific type of damage that I want to apply directly to the health, even if there is armour.
And the default value of this boolean is `false`, so it won't do anything unless you specify it as true.

**Comment by Crayder** [comment]
Yeah, I get that but it could be a bit more flexible...

**Comment by Jeroen52** [comment]
Can you explain how? It is already an optional boolean.

File: weapon-config.inc [hunk 1]
```diff
@@ -1200,7 +1200,7 @@ stock DamagePlayer(playerid, Float:amount, issuerid = INVALID_PLAYER_ID, weaponi
 		issuerid = INVALID_PLAYER_ID;
 	}
 
-	InflictDamage(playerid, amount, issuerid, weaponid, bodypart);
```
Comment (oscar-broman) [inline weapon-config.inc]: No need for a bool tag here

File: weapon-config.inc [hunk 2]
```diff
@@ -1200,7 +1200,7 @@ stock DamagePlayer(playerid, Float:amount, issuerid = INVALID_PLAYER_ID, weaponi
...
+	InflictDamage(playerid, amount, issuerid, weaponid, bodypart, bool:ignore_armour);
```
Comment (oscar-broman) [inline weapon-config.inc]: ..I mean here

File: weapon-config.inc [hunk 3]
```diff
@@ -4518,7 +4518,7 @@ static InflictDamage(playerid, Float:amount, issuerid = INVALID_PLAYER_ID, weapo
 		}
 	#endif
 
-	if (weaponid == WEAPON_COLLISION || weaponid == WEAPON_DROWN || weaponid == WEAPON_CARPARK) {
+	if (weaponid == WEAPON_COLLISION || weaponid == WEAPON_DROWN || weaponid == WEAPON_CARPARK || bool:ignore_armour) {
```
Comment (oscar-broman) [inline weapon-config.inc]: Also no need for a tag

**Comment by Crayder** [comment]
Can you explain how? It is already an optional boolean.

Sure. What if someone wanted to allow everything except bullets directly penetrate health? Or what if someone wanted to make a more realistic script and only only let the armor protect players from bullets to the torso? And surely armor isn't going to protect someone from flames.

**Comment by Jeroen52** [comment]
@Crayder That isn't something I thought of.
Wouldn't that need to be made completely differently something with `OnPlayerDamage`, right?

**Comment by oscar-broman** [comment]
I suppose we could make an array that decides if armour should be ignored, then make functions to change the array.

**Comment by Crayder** [comment]
I suppose we could make an array that decides if armour should be ignored, then make functions to change the array.

Yeah that would be great, it'd be like the per-weapon damage setting function.

I don't mind attempting to add this myself, I'll let Jeroen52 decide whether I or he should.

**Comment by Jeroen52** [comment]
I'm not too familiar with arrays, I can do the basic things like put data in arrays, read it, and delete it, but not a lot of different things.

If you want to, you can make it yourself.

**Comment by oscar-broman** [comment]
Separate functionality to modify the default behavior of armour/health will not conflict with this parameter, so I'll merge this and we'll keep this the way it is.
Feel free to make a pull request with the other suggestion.

**Comment by Crayder** [comment]
Please check https://github.com/Crayder/samp-weapon-config/commit/24486be7b0466d3ea35ad6b8e58b0ec7c5fe5709

I kind of worked quick, but I think it'd work. I haven't tested it.

**Comment by Crayder** [comment]
@Jeroen52 Check out pull 43

**Pull request merged**